src/main/java/com/triviamachine/triviamachine/controllers/ApplicaitonController.java

```diff
@@ -0,0 +1,98 @@
+package com.triviamachine.triviamachine.controllers;
+
+import com.triviamachine.triviamachine.database.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.annotation.PostConstruct;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+
+@Controller
+@RequestMapping
+public class ApplicaitonController {
+
+    @Autowired
+    private QuestionRepository questionRepo;
+    @Autowired
+    private QuestionScheduleRepository questionScheduleRepo;
+    @Autowired
+    private ResultsRepository resultsRepo;
+    @Autowired
+    AdminUserRepository adminRepo;
+
+    @Autowired
+    DateFormat dateFormat;
+
+    @GetMapping("/login")
+    public String getLogin(@AuthenticationPrincipal AdminUser user, Model model) {
+        model.addAttribute("user", user);
+        return "login";
+    }
+
+    @PostConstruct
+    public void ensureFallbackQuestionExists() {
+        List<Question> allQuestions = questionRepo.findAll();
+        if (allQuestions.size() == 0) {
+            Question fallback = new Question();
+            fallback.setQuestionText("Cake or Pie?");
+            fallback.setAnswerOne("Cake");
+            fallback.setAnswerTwo("Pie");
+            fallback.setAnswerThree("");
+            fallback.setAnswerFour("");
+            fallback.setCorrectAnswer((byte) 0);
```

Comment:
When in doubt, keep it simple.  Trying to optimize early almost always bites (bytes) you.

Instead:  As you gain experience, you'll learn where you _have_ to optimize and you'll naturally do those things in the future.